test/test_unary_ufuncs.py

```diff
@@ -58,8 +58,42 @@
                      -4988429.2, 4988429.2,
                      -1e20, 1e20)
 _float_extremals = (float('inf'), float('-inf'), float('nan'))
+_medium_length = 812
+_large_size = (1029, 917)
 
 
+# Returns generator of tensors of different sizes filled with values in domain
+# and with intested region filled with `vals`. This will help test different code
+# paths for the given vals
+def generate_tensors_from_vals(vals, device, dtype, domain):
+    offset = 63
+
+    assert _large_size[1] > (_medium_length + offset)  # large tensor should be large enough
+    assert len(vals) < _medium_length  # medium tensor should contain all vals
+    assert _medium_length % 4 == 0  # ensure vectorized code coverage
```

Comment:
Is my comment correct here?

Reply:
Kind of? Any contiguous innermost dimensions will be vectorized. This condition ensures there will be no non-vectorized tail.